tagstudio/src/qt/widgets/thumb_renderer.py

```diff
@@ -759,8 +760,26 @@ def _image_vector_thumb(self, filepath: Path, size: int) -> Image.Image:
             filepath (Path): The path of the file.
             size (tuple[int,int]): The size of the thumbnail.
         """
-        # TODO: Implement.
-        im: Image.Image = None
+
+        # Create an image to draw the svg to and a painter to do the drawing
+        image: QImage = QImage(size, size, QImage.Format.Format_ARGB32)
+        image.fill("#00000000")
+        painter: QPainter = QPainter(image)
+
+        # Create an svg renderer, then render to the painter
+        svg: QSvgRenderer = QSvgRenderer(str(filepath))
+        svg.render(painter)
+        del painter
```

Comment:
What is the reasons for the manual cleanup?

Reply:
I had done it to fix an error, however that error was also resolved by just ending the painter and I forgot to commit that.